qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -20,95 +24,132 @@
-		testPatient.givenName = "Yuan";
-		testPatient.middleName = "Youn";
-		testPatient.familyName = "Ching";
 		testPatient.gender = "Male";
-		testPatient.birthDay = "23";
+		testPatient.birthDay = "21";
 		testPatient.birthMonth = "May";
-		testPatient.birthYear = "1990";
-		registrationEditSectionPage.enterGivenName(testPatient.givenName);
-		registrationEditSectionPage.enterMiddleName(testPatient.middleName);
-		registrationEditSectionPage.enterFamilyName(testPatient.familyName);
+		testPatient.birthYear = "1992";
+		registrationEditSectionPage.enterGivenName(testPatient.givenName + prefix);
+		registrationEditSectionPage.enterMiddleName(testPatient.middleName + prefix);
+		registrationEditSectionPage.enterFamilyName(testPatient.familyName + prefix);
 		registrationEditSectionPage.selectPatientGender(testPatient.gender);
 		registrationEditSectionPage.clickOnBirthdateLabel();
 		registrationEditSectionPage.enterBirthDay(testPatient.birthDay);
 		registrationEditSectionPage.selectBirthMonth(testPatient.birthMonth);
 		registrationEditSectionPage.enterBirthYear(testPatient.birthYear);
 	}
+	
+	@And ("a user confirms the changes")
+	public void confirmChanges() throws InterruptedException {
+		registrationEditSectionPage.clickOnConfirmEdit();
+		dashboardPage = registrationEditSectionPage.confirmPatient();
+	}
 
 	@Then("the system saves the updated patient demographics")
 	public void systemSavesUpdatedDemographics() {
-		assertEquals(registrationSummaryPage.getPatientGivenName(), testPatient.givenName);
-		assertEquals(registrationSummaryPage.getPatientFamilyName(), testPatient.familyName);
+		assertThat(dashboardPage.getPatientGivenName(), is(testPatient.givenName + prefix));
+	}
+	
+	@And("a user clicks on the show contact downdrop")
+	public void clickOnShowContactDropDown() {
+		dashboardPage.clickOnShowContact();
 	}
 
-	@When("a user clicks on Edit link under contact info section from Registration Summary Page")
+	@And("a user clicks on Edit link under contact info section from Registration Summary Page")
 	public void loadContactInfoEditSection() {
-		registrationSummaryPage.clickOnShowContact();
-		registrationEditSectionPage = (RegistrationEditSectionPage) registrationSummaryPage.clickOnEditContact();
+		registrationEditSectionPage = dashboardPage.clickOnEditContact();
 	}
 
 	@Then("the system loads the edit contact information section")
 	public void systemLoadsEditContactInfoSection() {
-		assertPage(registrationEditSectionPage.waitForPage());
+		assertTrue(textExists("Contact Info"));
 	}
 
 	@And("a user edits contact information")
 	public void editContactInfo() {
-		testPatient.city = "Kyambogo";
-		testPatient.state = "Central";
-		testPatient.country = "Uganda";
-		testPatient.postalCode = "256";
-		testPatient.phone = "0785456654";
+		testPatient.city = "Adidas Abbeba";
+		testPatient.state = "Adidas Abbeba";
+		testPatient.country = "Ethiopia";
+		testPatient.postalCode = "3822";
+		testPatient.phone = "aaaaaaaaa";
```

Comment:
is there a country where citizens telephone numbers are in alphabetic letters?

Reply:
The alphabets are meant to test that the phone number only accepts numbers.

Reply:
> The alphabets are meant to test that the phone number only accepts numbers

This sounds to be xss vulnerability, @sherrif10 isn't ContactInfoTest supposed to be covered in [RATEST-308 ](https://issues.openmrs.org/browse/RATEST-308)?

Reply:
Nope thanks. they are quite different. with in this test , we are not handling vulnerability check the difference is between with in this test, we are not entering phone number with  any script element  tag for example <script>alert(0)</script>. which checks vulnerability. this test is supposed to be straight forward.